app/modules/sightings/resources.py

```diff
@@ -57,21 +80,182 @@ def get(self, args):
             'action': AccessOperation.WRITE,
         },
     )
-    @api.login_required(oauth_scopes=['sightings:write'])
+    # @api.login_required(oauth_scopes=['sightings:write'])
     @api.parameters(parameters.CreateSightingParameters())
-    @api.response(schemas.DetailedSightingSchema())
     @api.response(code=HTTPStatus.CONFLICT)
     def post(self, args):
         """
         Create a new instance of Sighting.
         """
+
+        data = {}
+        try:
+            data_ = json.loads(request.data)
+            data.update(data_)
+        except Exception:
+            pass
+
+        response = current_app.edm.request_passthrough(
+            'sighting.data', 'post', {'data': data}, ''
+        )
+
+        response_data = None
+        result_data = None
+        if response.ok:
+            response_data = response.json()
+            result_data = response_data.get('result', None)
+
+        if (
+            not response.ok
+            or not response_data.get('success', False)
+            or result_data is None
+        ):
+            log.warning('Sighting.post failed')
+            passed_message = {'message': {'key': 'error'}}
+            if response_data is not None and 'message' in response_data:
+                passed_message = response_data['message']
+            abort(success=False, passed_message=passed_message, message='Error', code=400)
+
+        # if we get here, edm has made the sighting.  now we have to consider encounters contained within,
+        # and make houston for the sighting + encounters
+
+        # FIXME must cleanup edm cruft!
+        if ('encounters' in data and 'encounters' not in result_data) or (
+            'encounters' not in data and 'encounters' in result_data
+        ):
+            log.error(
+                'Sighting.post missing encounters in one of %r or %r'
+                % (data, result_data)
+            )
+            _cleanup_post_and_abort(
+                None, None, 'Missing encounters between data and result'
+            )
+        if not len(data['encounters']) == len(result_data['encounters']):
+            log.error(
+                'Sighting.post imbalanced encounters in %r or %r' % (data, result_data)
+            )
+            _cleanup_post_and_abort(
+                None, None, 'Imbalance in encounters between data and result'
+            )
+        # i think this was official declared as law today 2021-02-24
```

Comment:
So why not check it before talking to the EDM at all?

Reply:
i would argue not _only_ should that happen, but (or instead of?) i should have the edm *fail* when no encounters are passed as well.  so it never should get here -- either way!

Reply:
moved this way up top in function now; before the passthrough call to edm.